### PR TITLE
Allow custom image and args

### PR DESCRIPTION
**Pull request by AlexisDuf**
##### SUMMARY
Being able to customize Image and Args from the container group pod specification 

related #10568 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME

 - API (awx/main/tasks.py)

##### AWX VERSION

```
awx: 19.2.2
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: awx/main/tasks.py [hunk 1]
```diff
@@ -3088,8 +3088,8 @@ def pod_definition(self):
             pod_spec_override = parse_yaml_or_json(self.task.instance.instance_group.pod_spec_override)
         pod_spec = {**default_pod_spec, **pod_spec_override}
 
-        pod_spec['spec']['containers'][0]['image'] = ee.image
-        pod_spec['spec']['containers'][0]['args'] = ['ansible-runner', 'worker', '--private-data-dir=/runner']
+        if pod_spec['spec']['containers'][0]['image'] is None or pod_spec['spec']['containers'][0]['image'] == "":
```
Comment (shanemcd) [inline awx/main/tasks.py:3091]: This feels wrong to me. 

I think the right thing to do would change the `image:` field in the default pod spec to say something like:

 ```
image: <placeholder> # Value comes from the resolved Execution Environment at runtime
```

File: awx/main/tasks.py [hunk 2]
```diff
@@ -3088,8 +3088,8 @@ def pod_definition(self):
             pod_spec_override = parse_yaml_or_json(self.task.instance.instance_group.pod_spec_override)
         pod_spec = {**default_pod_spec, **pod_spec_override}
 
-        pod_spec['spec']['containers'][0]['image'] = ee.image
-        pod_spec['spec']['containers'][0]['args'] = ['ansible-runner', 'worker', '--private-data-dir=/runner']
```
Comment (shanemcd) [inline awx/main/tasks.py:3092]: -1. See https://github.com/ansible/awx/issues/10568#issuecomment-872314993

**Review by shanemcd — CHANGES_REQUESTED** [review]
Comments inline.

File: awx/main/tasks.py (hunk 1 above)
Comment (nmaupu) [reply awx/main/tasks.py:3091]: If I override the pod spec's image, I would think my configuration would be taken into account. Changing the EE to actually change the image is confusing.

File: awx/main/tasks.py (hunk 2 above)
Comment (nmaupu) [reply awx/main/tasks.py:3092]: It's fine to put a default value if not set but I feel it's wrong when it's silently replaced.
User should be able to override to whatever value they'd like

**Comment by shanemcd** [comment]
The default pod spec now contains the command that gets run.

**Pull request closed without merging**